src/protocol/codecs/msgpack.rs

```diff
@@ -382,6 +462,181 @@ impl<'de> Deserialize<'de> for HashDigest {
     }
 }
 
+/// An Ed25519 Signature.
+#[derive(Copy, Clone, Eq)]
+pub struct Ed25519Signature(pub [u8; 64]);
+
+impl Debug for Ed25519Signature {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("Signature").field(&self.0.to_vec()).finish()
+    }
+}
+
+impl PartialEq for Ed25519Signature {
+    fn eq(&self, other: &Self) -> bool {
+        for i in 0..64 {
+            if self.0[i] != other.0[i] {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+impl Serialize for Ed25519Signature {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_bytes(&self.0[..])
+    }
+}
+
+impl<'de> Deserialize<'de> for Ed25519Signature {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(Ed25519Signature(
+            deserializer.deserialize_bytes(SignatureVisitor)?,
+        ))
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Clone, Deserialize)]
+pub struct MultisigSignature {
+    #[serde(rename = "subsig")]
+    pub subsigs: Vec<MultisigSubsig>,
+
+    #[serde(rename = "thr")]
+    pub threshold: u8,
+
+    #[serde(rename = "v")]
+    pub version: u8,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Deserialize)]
+pub struct MultisigSubsig {
+    #[serde(rename = "pk")]
+    pub key: Ed25519PublicKey,
+
+    #[serde(rename = "s")]
+    pub sig: Option<Ed25519Signature>,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct Ed25519PublicKey(pub [u8; 32]);
```

Comment:
32 could be const to avoid magic numbers.
Lack of doc comment (///) like in other structures.